pkg/pgmodel/pgx.go

```diff
@@ -455,27 +342,19 @@ func runInserterRoutine(conn pgxConn, input chan insertDataRequest) {
 			continue
 		}
 
-		startHandling := time.Now()
 	hotReceive:
-		for {
-			for i := 0; i < 1000; i++ {
-				receivingNewReqs := handler.nonblockingHandleReq()
-				if !receivingNewReqs {
-					break hotReceive
-				}
-			}
-			if time.Since(startHandling) > flushTimeout {
-				handler.flushTimedOutReqs()
-				startHandling = time.Now()
+		for handler.nonblockingHandleReq() {
+			if len(handler.pending.batch.sampleInfos) >= flushSize {
+				break hotReceive
```

Comment:
We should benchmark sleeping when we have data, instead of relying on the hot-loop; I suspect that the individual workers won't have enough work for the loop to be justified now that each one only handles a single metric. (this can go in the next PR)